crypto/merklearray/merkle.go

```diff
@@ -31,15 +31,15 @@ type Tree struct {
 	_struct struct{} `codec:",omitempty,omitemptyarray"`
 
 	// Level 0 is the leaves.
-	Levels []Layer            `codec:"lvls,allocbound=-"`
+	Levels []Layer            `codec:"lvls,allocbound=9999"`
 	Hash   crypto.HashFactory `codec:"hsh"`
 }
 
 // Proof contains the merkle path, along with the hash factory that should be used.
 type Proof struct {
 	_struct struct{} `codec:",omitempty,omitemptyarray"`
 
-	Path        []crypto.GenericDigest `codec:"pth,allocbound=-"`
+	Path        []crypto.GenericDigest `codec:"pth,allocbound=9999"`
```

Comment:
Please add a constant for this one. The actual number should be much smaller than 9999. Most likely 256 is a good limit.

Reply:
@algoidan is looking into this.

Reply:
it looks like that if we want a proof for all the leaves in the tree , this might contain all the hashes in the tree
since we bound the number of leaves to 2^16, we should bound this to 2^17

Reply:
Tsachi thinks this should be in the order of the hight of the tree.

Reply:
The final number I came out with is 2^15